cpp/search/search.h

```diff
@@ -282,8 +282,10 @@ struct Search {
   void printRootEndingScoreValueBonus(std::ostream& out) const;
 
   //Get detailed analysis data, designed for lz-analyze and kata-analyze commands.
-  void getAnalysisData(std::vector<AnalysisData>& buf, int minMovesToTryToGet, bool includeWeightFactors, int maxPVDepth) const;
   void getAnalysisData(const SearchNode& node, std::vector<AnalysisData>& buf, int minMovesToTryToGet, bool includeWeightFactors, int maxPVDepth) const;
+  void getAnalysisData(std::vector<AnalysisData>& buf, int minMovesToTryToGet, bool includeWeightFactors, int maxPVDepth) const;
+  void getExtendedAnalysisData(const SearchNode& node, std::vector<AnalysisData>& buf,AnalysisData& nodeData, float (&policyProbs)[NNPos::MAX_NN_POLICY_SIZE], int minMovesToTryToGet, bool includeWeightFactors, int maxPVDepth) const;
```

Comment:
Rather than add a whole new function, why not simply have analysis.cpp call this already-existing function?
```bool getRootValues(ReportedSearchValues& values) const;  //search.h```

Also, the following three-liner should be able to get the full policy distribution from the bot., without needing to modify the code in the searchresults.cpp.
```
float policyProbs[NNPos::MAX_NN_POLICY_SIZE];
assert(search->rootNode != NULL && search->rootNode->nnOutput != nullptr); //Should always be populated after a search with at least 1 visit.
std::copy(search->rootNode->nnOutput->policyProbs, search->rootNode->nnOutput->policyProbs+NNPos::MAX_NN_POLICY_SIZE, policyProbs);
```

Maybe this is abstraction-breaking a little for the Search struct, so could warrant a function to do this, but it's also not worse than similar things that are already done elsewhere in some parts of the code, so I'd be okay with it.